### PR TITLE
Set up global audio player

### LINKED ISSUE #589: Implement global audio player on mobile audio results view
## Problem

The mobile audio view needs the global player implemented. 

Figma: https://www.figma.com/file/w60dl1XPUvSaRncv1Utmnb/Openverse-Releases?node-id=133%3A13723

## Description

As far as I know, this only displays on mobile and on the `/search/audio` endpoint. 

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #589 by @zackkrida 

## Description

This PR
- creates the global audio section as a part of the layout
  - this section renders the audio track in the 'Global' layout for mobile screens
- removes the defunct `VAudioController` component (as there is no longer `<audio>` in it)
  - those duties have been migrated to `VAudioTrack` itself
- cleans up and refactors code to keep the flow of control between `VPlayPause`, `VWaveform`, `VAudioTrack`, Vuex active media store and `<audio>` sane
- adds more sample music data to make testing pleasant 

## Testing Instructions

0. Checkout the branch and run the dev server.
1. Resize the screen to a mobile layout using dev tools, bonus points for testing on an actual phone.
2. Play a song. The global audio layout should appear and work.
3. Change it midway. The new song should immediately play and the play-pause icons should be correct for both songs.
4. Pause the song and play again. It should pick up where it paused.
5. Pause from the global layout and play from the search result icon and vice versa. It should work.
6. Close the player when the song is playing. It should pause and the global layout should disappear.
7. Keep the API running on port 8000 for cover art. Images should show up in the right places.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by zackkrida** [comment]
@dhruvkb this looks so awesome! I would suggest basing this branch on the `redesign` branch, as soon as possible, there's likely significant merge conflicts to work through.

**Comment by dhruvkb** [comment]
Rebased on top of `redesign`, fortunately there weren't any big merge conflicts.

**Review by zackkrida — CHANGES_REQUESTED** [review]
Amazing work, I noticed a few bugs:

- clicking anywhere on the global audio player waveform resets playback to the start of the track, instead of the clicked timestamp
- selecting a single image result keeps the global player up, but pauses the track and causes a state mismatch where the button shows the pause icon but the track isn't playing

**Comment by dhruvkb** [comment]
@zackkrida
- This seems to be a Chrome bug that appears when the audio URL response does not contain the Content-Size header. This only happens for a combination of Chrome browser + Jamendo track. Could you try with Firefox and Safari?
- The track gets paused because it's a different layout (#599 should fix this). I'll look into the state mismatch.

**Comment by zackkrida** [comment]
@dhruvkb you're correct on both accounts, thanks!

**Comment by dhruvkb** [comment]
@zackkrida I tested this with #599 and the state mismatch seems to go away on it's own because now there's no layout change 🎉!

**Review by zackkrida — APPROVED** [review]
Approving, we can figure out how to deal with the Jamendo API issue in the future.

**Comment by dhruvkb** [comment]
X-referencing @obulat's bug report from https://github.com/WordPress/openverse-frontend/pull/605#pullrequestreview-852932645.

**Review by obulat — APPROVED** [review]
This is amazing, Dhruv! 🎉 🎆 

Sorry for replacing the 'small' svg icon - I didn't know we needed the small version, too.

There are a couple of things I noticed:
1. Is the player supposed to be open on single audio result page? When I click on an audio link from the search result page, the single audio result page opens, and the global player is still there. However, when I click on the play button on the page, the player disappears, and the main audio starts to play.
When I click `back` from the single audio result page, that song keeps playing, even though it's not in the results. And if you look at the results, you might think that no audio is playing - all are in paused state. ~And if you click on `play` this time, the global audio player doesn't appear~ This happened once, and I couldn't reproduce it.
I tested on Safari, Chrome, and Firefox, and it works great!
By the way, when you click 'Play' on Safari, you don't get the pink outline around the play button

File: test/unit/specs/components/AudioTrack/audio-track.spec.js [hunk 1]
```diff
@@ -16,9 +17,39 @@ const useVueI18n = (vue) => {
     i18n,
   }
 }
+const useVuexStore = (vue) => {
```
Comment (obulat) [inline test/unit/specs/components/AudioTrack/audio-track.spec.js:20]: Nice abstraction! Hopefully we can create similar abstraction for all tests that use testing-library, a utility similar to `render` that we had previously.

**Pull request merged**